homeassistant/components/xiaomi_miio/number.py

```diff
@@ -0,0 +1,214 @@
+    FEATURE_SET_MOTOR_SPEED: NumberType(
+        name="Motor Speed",
+        icon="mdi:fast-forward-outline",
+        short_name=ATTR_MOTOR_SPEED,
+        unit_of_measurement="rpm",
+        min=200,
+        max=2000,
+        step=10,
+        service=SERVICE_SET_MOTOR_SPEED,
+        available_with_device_off=False,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        model = config_entry.data[CONF_MODEL]
+        device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierNumber
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierNumber
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierNumber
+        else:
+            return
+
+        for feature in NUMBER_TYPES:
+            number = NUMBER_TYPES[feature]
+            if feature & device_features and feature in NUMBER_TYPES:
+                entities.append(
+                    entity_class(
+                        f"{config_entry.title} {number.name}",
+                        device,
+                        config_entry,
+                        f"{number.short_name}_{config_entry.unique_id}",
+                        number,
+                        coordinator,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class XiaomiNumber(XiaomiCoordinatedMiioEntity, NumberEntity):
+    """Representation of a generic Xiaomi attribute selector."""
+
+    def __init__(self, name, device, entry, unique_id, number, coordinator):
+        """Initialize the generic Xiaomi attribute selector."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        self._state = None
+        self._attr_icon = number.icon
+        self._attr_unit_of_measurement = number.unit_of_measurement
+        self._attr_min_value = number.min
+        self._attr_max_value = number.max
+        self._attr_step = number.step
+        self._supported_features = 0
+        self._device_features = 0
+        self._state_attrs = {}
+        self._controller = number
+        self._enum_class = None
+        self._attributes = None
+        self._value = None
+
+    @property
+    def value(self):
```

Comment:
We can use `self._attr_value` instead.

Reply:
No we can not. We need to cache the value if it is change with a command. It later will be refreshed if the coordinator supplies the update.

Reply:
I don't see the problem. Please explain more.

Reply:
It takes some time before the update from the humidifier comes back. We use the value attribute to store the temporary cached state if a device command was successful.

Reply:
Why can't we use `self._attr_value` for that purpose as well?

Reply:
Okay, testing right now

Reply:
Updated the code as requested, removed the value property code.